scripts/nix-profile-daemon.sh.in

```diff
@@ -2,21 +2,31 @@
 if [ -n "$__ETC_PROFILE_NIX_SOURCED" ]; then return; fi
 __ETC_PROFILE_NIX_SOURCED=1
 
+NIX_USER_PROFILE_DIR="@localstatedir@/nix/profiles/per-user/$USER"
+
+# Set up the per-user profile.
+mkdir -m 0755 -p "$NIX_USER_PROFILE_DIR"
+if ! test -O "$NIX_USER_PROFILE_DIR"; then
+    echo "ERROR: bad ownership on $NIX_USER_PROFILE_DIR" >&2
```

Comment:
Can you give some more guidance to the user about why this might have happened and what they can do?